### PR TITLE
[soundtouch] update to 2.3.1

### PR DESCRIPTION
**Pull request by Be-ing**
**Describe the pull request**
I worked with upstream to implement a CMake build system, so I
removed the CMakeLists.txt from this port.

- #### What does your PR fix?  
   * Headers were not installed by the port before.
  * The port did not allow dynamic linking before.
  * pkgconfig and CMake config files were not installed before.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all but UWP

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

File: ports/soundtouch/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "soundtouch",
+  "version-string": "2.3.1",
```
Comment (JonLiu1993) [inline ports/soundtouch/vcpkg.json:3]:
```suggestion
  "version-semver": "2.3.1",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@Be-ing ,Have you tested the features locally? I tested the features in x86-windows and x64-windows but all failed:
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/7119349/install-x86-windows-dbg-out.log)

**Comment by Be-ing** [comment]
It seems the soundtouchdll feature only does not work on Windows with static linking.

**Comment by Be-ing** [comment]
Upstream merge request to fix SoundTouchDLL build with MSVC: https://gitlab.com/soundtouch/soundtouch/-/merge_requests/18

**Comment by Be-ing** [comment]
Upstream merged my fix for the soundtouchdll feature and moved the 2.3.1 tag to include it. I updated the hash.

**Comment by JonLiu1993** [comment]
@Be-ing ,All feature are tetsed successfully in the following triplet:
- x86-windows
- x64-windows
- x64-linux 
but in x64-windows-static triplet failed:
````
DLLs should not be present in a static build, but the following DLLs were found:

    F:/test-pr/1/vcpkg/packages/soundtouch_x64-windows-static/bin/SoundTouchDLL.dll
    F:/test-pr/1/vcpkg/packages/soundtouch_x64-windows-static/debug/bin/SoundTouchDLL.dll

There should be no bin\ directory in a static build, but F:\test-pr\1\vcpkg\packages\soundtouch_x64-windows-static\bin is present.
There should be no debug\bin\ directory in a static build, but F:\test-pr\1\vcpkg\packages\soundtouch_x64-windows-static\debug/bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()

Found 2 error(s). Please correct the portfile:
````

**Comment by Be-ing** [comment]
SoundTouchDLL is only meant to be built as a DLL, hence the name. It is hardcoded in the upstream build system as a shared library. So if that warning is a problem, the only solution I think would be to disable the soundtouchdll feature for static builds.

**Comment by JonLiu1993** [comment]
@Be-ing ,Could you please fix the x64-windows-static triplet error and handle the Conflicting files?

**Comment by Be-ing** [comment]
I need an answer how to handle the interaction between the soundtouchdll feature and static linking. I can think of a few options:

1. Fatal error if soundtouchdll is enabled with static linking.
2. Nonfatal warning if soundtouchdll is enabled with static linking.
3. Allow building soundtouchdll with static linking and silence the warning from vcpkg.

I don't particularly care what the answer is; I don't have a need for SoundTouchDLL. It was made to implement Delphi and Pascal bindings for SoundTouch, but SoundTouch is primarily intended to be used as a C++ library.

File: ports/soundtouch/portfile.cmake [hunk 2]
```diff
@@ -1,31 +1,33 @@
-vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY ONLY_DYNAMIC_CRT)
```
Comment (JackBoosY) [inline ports/soundtouch/portfile.cmake:1]: SoundTouchDLL is only meant to be built as a DLL, hence the name. It is hardcoded in the upstream build system as a shared library.
Therefore, I think we should keep this line.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
> I need an answer how to handle the interaction between the soundtouchdll feature and static linking. I can think of a few options:
> 
> 1. Fatal error if soundtouchdll is enabled with static linking.
> 2. Nonfatal warning if soundtouchdll is enabled with static linking.
> 
> I don't particularly care what the answer is; I don't have a need for SoundTouchDLL. It was made to implement Delphi and Pascal bindings for SoundTouch, but SoundTouch is primarily intended to be used as a C++ library.

Users can use different triplet or `VCPKG_CRT_LINAKGE` settings to synchronize with their projects.

File: ports/soundtouch/portfile.cmake (hunk 2 above)
Comment (Be-ing) [reply ports/soundtouch/portfile.cmake:1]: I think you are misunderstanding. SoundTouchDLL is an optional feature. It is a C wrapper around the core C++ library.

**Comment by JonLiu1993** [comment]
After finish your suggesstion changes,please run command "./vcpkg x-add-version  soundtouch --overwrite-version" then commit the changes again

**Comment by JackBoosY** [comment]
Is there anything else I should help you with?

**Comment by Be-ing** [comment]
@JonLiu1993 I'm confused. Is there something more that needs to be done here? You removed then re-added the "requires:author-response" tag. I have taken care of updating the versions files.

**Comment by Be-ing** [comment]
> Is there anything else I should help you with?

I'm not sure if this was directed at me or @JonLiu1993. I don't think there's anything else to do on this PR. On a tangential note, it would be great to finish up #17111.

**Comment by JonLiu1993** [comment]
@Be-ing ,Sorry,I want to know if you have completed all the work. I need to test the feature locally. I'm sorry for the confusion you caused.

**Comment by JonLiu1993** [comment]
All feature are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-linux

**Review by JonLiu1993 — APPROVED** [review]

File: ports/soundtouch/vcpkg.json (hunk 1 above)
Comment (BillyONeal) [reply ports/soundtouch/vcpkg.json:3]: Maybe it should be just "version" instead of "version-semver". It certainly shouldn't be version-string.

**Comment by BillyONeal** [comment]
[version-patch.patch.txt](https://github.com/microsoft/vcpkg/files/7180006/version-patch.patch.txt)

version-string is certainly wrong. If upstream follows semantic versioning then please apply JonLiu's suggestion, otherwise apply this patch.

**Comment by JonLiu1993** [comment]
@Be-ing ,Please confirm If upstream follows semantic versioning. and if no please change version-semver to version.the PR was created without permissions for me to fix it

**Comment by Be-ing** [comment]
Yes upstream follows semver.

**Comment by JonLiu1993** [comment]
@Be-ing ,Please merge master,

**Review by JackBoosY — APPROVED** [review]
Did you test the usage?

File: ports/soundtouch/vcpkg.json [hunk 3]
```diff
@@ -1,14 +1,29 @@
 {
   "name": "soundtouch",
-  "version-string": "2.0.0",
-  "port-version": 7,
+  "version-semver": "2.3.1",
```
Comment (JonLiu1993) [inline ports/soundtouch/vcpkg.json:3]: The developer confirms it is following semantic versioning?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/soundtouch/vcpkg.json (hunk 3 above)
Comment (Be-ing) [reply ports/soundtouch/vcpkg.json:3]: I do not want to bother upstream again by asking this. Do you really expect every upstream to explicitly say "I follow semver" to call it semver in vcpkg?

Comment (BillyONeal) [reply ports/soundtouch/vcpkg.json:3]: If they don't document that they follow semver, they don't, so this should be `"version"`.

Comment (Be-ing) [reply ports/soundtouch/vcpkg.json]: okay, I changed it to `version`

**Comment by BillyONeal** [comment]
Thanks for the help working with upstream!

**Comment by Be-ing** [comment]
One less port in the overlays! :)

**Pull request merged**